main.go

```diff
@@ -219,3 +203,44 @@ func encodeConfig(cfg *configv1alpha1.Configuration) (string, error) {
 	}
 	return buf.String(), nil
 }
+
+func apply(configFile string) (ctrl.Options, configv1alpha1.Configuration) {
+	var err error
+	options := ctrl.Options{
+		Scheme: scheme,
+	}
+	config := configv1alpha1.Configuration{}
+	if configFile != "" {
+		options, err = options.AndFrom(ctrl.ConfigFile().AtPath(configFile).OfKind(&config))
+		if err != nil {
+			setupLog.Error(err, "unable to load the config file")
+			os.Exit(1)
+		}
+
+		cfgStr, err := encodeConfig(&config)
+		if err != nil {
+			setupLog.Error(err, "unable to encode config file")
+			os.Exit(1)
+		}
+		setupLog.Info("Successfully loaded config file", "config", cfgStr)
+	}
+
+	setOptionDefaults(&options)
```

Comment:
are you sure this can't be done before calling `AndFrom`?
Or are you just trying to be consistent with the function below?

Reply:
If we set default values in options before calling `AndFrom`, we can not overwrite them in `Configuration`, e.g. currently our default port is 9443, even if we set this to 9444 in `Configuration`, we'll still get port 9443. This is because in `AndFrom`, we'll check where port is zero value, if not, we'll not update it. see https://github.com/kubernetes-sigs/controller-runtime/blob/15154aaa67679df320008ed45534f83ff3d6922d/pkg/manager/manager.go#L480-L482

Reply:
I have no idea why we can't preview these linked codes.